interface/modules/custom_modules/oe-module-comlink-telehealth/src/Repository/TeleHealthUserRepository.php

```diff
@@ -0,0 +1,153 @@
+            } else {
+                $binds[] = (new \DateTime())->format(DATE_ISO8601);
+            }
+        }
+
+        return QueryUtils::sqlInsert($sql, $binds);
+    }
+
+    public function getUser($username): ?TeleHealthUser
+    {
+        $result = $this->search(['username' => $username]);
+        if ($result->hasData()) {
+            return $result->getData()[0];
+        }
+        return null;
+    }
+
+    protected function createResultRecordFromDatabaseResult($row)
+    {
+        $dateFormat = "Y-m-d H:i:s";
+        $user = new TeleHealthUser();
+        $user->setId($row['id'])
+            ->setUsername($row['username'])
+            ->setAuthToken($row['auth_token'])
+            ->setDbRecordId($row['patient_id'] ?? $row['user_id'])
+            ->setIsPatient(isset($row['patient_id']))
+            ->setIsActive($row['active'] == 1)
+            ->setRegistrationCode($row['app_registration_code']);
+
+        if (isset($row['date_registered'])) {
+            $date = \DateTime::createFromFormat($dateFormat, $row['date_registered']);
+            if ($date !== false) {
+                $user->setDateRegistered($date);
+            } else {
+                $this->logger->errorLogCaller('failed to create date_registered', ['value' => $row['date_registered']]);
+            }
+        }
+        if (isset($row['date_created'])) {
+            $date = \DateTime::createFromFormat($dateFormat, $row['date_created']);
+            if ($date !== false) {
+                $user->setDateCreated($date);
+            } else {
+                $this->logger->errorLogCaller('failed to create date_created', ['value' => $row['date_created']]);
+            }
+        }
+        if (isset($row['date_updated'])) {
+            $date = \DateTime::createFromFormat($dateFormat, $row['date_updated']);
+            if ($date !== false) {
+                $user->setDateUpdated($date);
+            } else {
+                $this->logger->errorLogCaller('failed to create date_updated', ['value' => $row['date_updated']]);
+            }
+        }
+        return $user;
+    }
+
+
+    /**
+     * Since users can't see this password and they can't change this password w/o system user intervention we are just
+     * going to generate a random uuid for the password for the user.  The algorithm can be changed in the future in
+     * the event there is a compromise and we can use the registration API to change it.  Originally we wanted to use
+     * the user's password, however, we have no easy way that OpenEMR gives us access to the password and talking with
+     * the OpenEMR Administrators the hashing algorithm can be changed on the fly w/ OpenEMR so using the hash is not a
+     * good idea either.  Since the business requirement is that users can't see their password we will generate the
+     * password and use our standard encrypt method just in case the DB is taken.
+     */
```

Comment:
there's also a random generator function that you can use to create any length of string with whatever characters you want:
https://github.com/openemr/openemr/blob/master/src/Common/Utils/RandomGenUtils.php#L34

Reply:
yea, I've used that in another spot.  I figured the UUID was fine here as it pretty random.